src/ApiService/ApiService/TestHooks/IpOperationsTestHooks.cs

```diff
@@ -28,10 +28,13 @@ public class IpOperationsTestHooks {
             var name = query["name"];
 
             var nic = await _ipOps.GetPublicNic(rg, name);
-
-            var resp = req.CreateResponse(HttpStatusCode.OK);
-            await resp.WriteStringAsync(nic.Get().Value.Data.Name);
-            return resp;
+            if (nic != null) {
+                var resp = req.CreateResponse(HttpStatusCode.OK);
+                await resp.WriteStringAsync(nic.Get().Value.Data.Name);
```

Comment:
i think calling nic.get here is querying the server twice. 
The data should already be available in Value.Data.

Same for ip.Get below

Reply:
also if you have to use the get method use the async version